artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/client/SendAcknowledgementHandler.java

```diff
@@ -41,4 +41,13 @@
     * @param message message sent asynchronously
     */
    void sendAcknowledged(Message message);
+
+   default void sendFailed(Message message, Exception e) {
+      /**
+       * By default ignore failures to preserve compatibility with existing implementations.
+       * If the message makes it to the broker and a failure occurs sendAcknowledge() will
```

Comment:
Is this actually true? From skimming the diff it seems like it may not be as there are some if-else sections added that call one of the handler methods only. Unless somewhere else is also calling the sendAcknowledged method (possibly in duplicate?) then it seems if only the failure method gets called, and this default method drops it, that would mean an existing listener that doesn't implement sendFailed then wont see anything at all now. IF that is the case it suggests some tests are missing.

Should this default method impl perhaps be calling sendAcknowledged()? (uglier alternative, the calling code could check the sendFailed method is actually implemented and decide which to call)

Reply:
Ill let @jbertram comment here in original version it was done but for reason justin changed it. Im sure for good reason

Reply:
@gemmellr, at this point I'd say work up a test and send a PR. IMO, there's not much use discussing this on a PR that's already been merged.

Reply:
I'm not going digging with tests, I just thought I'd point out the apparently incorrect comment added, and seeming change of behaviour to what it comments on. If you can't see an issue thats fine.